include/openenclave/internal/raise.h

```diff
@@ -85,20 +93,64 @@ OE_EXTERNC_BEGIN
 // variadic paramter is empty in OE_RAISE_MSG(RESULT, fmt, ...)
 // eg : OE_RAISE_MSG(OE_FAILURE, "your message", NULL);
 
-#define OE_RAISE_MSG(RESULT, fmt, ...)   \
-    do                                   \
-    {                                    \
-        result = (RESULT);               \
-        if (result != OE_OK)             \
-        {                                \
-            OE_TRACE_ERROR(              \
-                fmt " (oe_result_t=%s)", \
-                ##__VA_ARGS__,           \
-                oe_result_str(result));  \
-        }                                \
-        goto done;                       \
+/* Note: on Linux, the above example fails, as the NULL pointer is fed as the
+ * value for oe_result_t in log, showing invalid (null) output.
+ *
+ * A solution is to define OS specfic versions of OE_RAISE_MSG() that
```

Comment:
Is it a simpler solution to simply reverse the order of the strings to make it consistent with `OE_RAISE()`?

```c
#define OE_RAISE_MSG(RESULT, fmt, ...)                                     \
    do                                                                     \
    {                                                                      \
        result = (RESULT);                                                 \
        if (result != OE_OK)                                               \
        {                                                                  \                                                          \
            OE_TRACE_ERROR(                                                \
                ":%s | " fmt,                                              \
                oe_result_str(result),                                     \
                ##__VA_ARGS__);                                            \
        }                                                                  \
        goto done;                                                         \
    } while (0)
```

Reply:
I tested with reversing the order of the strings as you mentioned, and the macro still fails, though in a different way.

If you look at the definition of OE_TRACE_ERROR(), you would see that the NULL ##__VA_ARGS__ would be fed to the first "%s" of its format (fmt " [%s:%s:%d]\n"), since "fmt" does not need any input. This also result in __FILE__, __FUNCTION__, and __LINE__ to be misaligned in the log line.